framework/src/node/consensus/certificate_generation/utils.ts

```diff
@@ -85,3 +85,17 @@ export const verifyAggregateCertificateSignature = (
 		threshold,
 	);
 };
+
+export const getSortedWeightsAndValidatorKeys = (
+	validatorKeysWithWeightsParam: {
+		weight: bigint;
+		blsKey: Buffer;
+	}[],
+) => {
+	const validatorKeysWithWeights = validatorKeysWithWeightsParam.map(o => ({ ...o }));
```

Comment:
instead of creating new function, sort `validatorKeysWithWeights` once, and loop once to generate weight and validatorKeys array looks more efficient. ex)
```
validatorKeysWithWeights.sort(xxx);
weights = [];
validatorKeys = [];
for (x of validatorKeysWithWeights)
  weights.push()
  validatorKeys.push()
```

Reply:
minor comment but prefer a more descriptive variable name for `o`

Reply:
done